### ISSUE TITLE
New computational gridding

### ISSUE THREAD

**Issue by dhardestylewis** [dhardestylewis/terrain_aggregator#71]
# tldr; new gridding scheme
- 44 hours to retile each new subgrid (40 hrs wait + 4 hrs compute)
- new gridding will be 414x512 subgrids
- each new subgrid will have ~157x194 tiles
- each new tile will have 256x256 pixels
- each new tile will have 16 pixel buffer

# ref; old gridding scheme
- 120 hours to retile each old subgrid (0 hrs wait + 120 hrs compute)
- old gridding had 16x17 subgrids
- each old subgrid had ~166x166 tiles
- each old tile had 1600x1600 pixels
- each old tile had 100 pixel buffer

# scratch sheet for reference to set up new griddings on different systems

50 jobs available max in Stampede2's `normal` queue (wait-time 40 hours)
- 2 jobs left open for `long` queue (wait-time 7 seconds)
= 48 jobs on Stampede2 `normal` queue

64 maximum recommended cores per node on `normal` queue

69 maximum available nodes per job (topping off at requesting 69*48=3312 nodes out of 3360 total `normal` nodes available -- we want fewer because there will likely be a small number of nodes down at any given time, ~10)

48 jobs * 64 cores per node * 69 nodes per job = 211968 cores available, one for each separate computational grid

sqrt(211968)~=460, prime factorization of 211968 is 23*3^2*2^10.

414=23*3^2*2
512=2^9
these are the closest divisors of 211968 to 460 that multiply to 211968

so computational grid will be 414x512 (because Texas is slightly taller than wide)

If new total computational envelope of Texas is similar to previous envelope and adjusting for the smaller width x height of each new tile, there will be an estimated 157x194 tiles within each subgrid.
(The new envelope will be slightly larger because it is derived from HUC8buffered100m intersect Texas and the old one was HUC12buffered100m intersect Texas -- HUC8s are larger than HUC12s)

Previous gridding took 120 hours to retile 71 billion pixels within each subgrid, new gridding estimated to take 4 hours to retile 2 billion pixels within each new subgrid

**Comment by dhardestylewis** [comment]
memory should be less of an issue this round because we are running roughly half as many subgrids per node in parallel as previous run. Previous run lost 7 subgrids to memory errors

**Comment by dhardestylewis** [comment]
estimated to cost 14,000 SUs

**Comment by dhardestylewis** [comment]
these are just the data tiles, not viz tiles

**Comment by dhardestylewis** [comment]
align tiles with quarter quads

**Comment by dhardestylewis** [comment]
# at 1m gridding by request from [@TNRIS](https://github.com/TNRIS)

## following previous estimate above and adjusting from .25m basemap to 1m basemap

# 1m basemap gridding scheme
- new gridding will be 460x460 subgrids
- each new subgrid will have ~44x44 tiles
- each new tile will have 256x256 pixels
- each new tile will have 0 pixel buffer
